src/collectors/python.d.plugin/puppet/puppet.chart.py

```diff
@@ -75,7 +75,6 @@ def __init__(self, configuration=None, name=None):
         UrlService.__init__(self, configuration=configuration, name=name)
         self.order = ORDER
         self.definitions = CHARTS
-        self.url = 'https://{0}:8140'.format(socket.getfqdn())
```

Comment:
Let's change it to

```python
self.url = self.configuration.get('url', 'https://{0}:8140'.format(socket.getfqdn()))
```

Reply:
Good point, that keeps the default behaviour while allowing to actually use the `url` chart parameter.
Pushed change and verified on puppet test infra.